numba/np/arraymath.py

```diff
@@ -4836,3 +4843,169 @@ def impl(filt, trim='fb'):
+                    mask &= (ar1 != a)
+            else:
+                mask = np.zeros(len(ar1), dtype=np.bool_)
+                for a in ar2:
+                    mask |= (ar1 == a)
+            return mask
+
+        # Otherwise use sorting
+        if not assume_unique:
+            # Equivalent to ar1, inv_idx = np.unique(ar1, return_inverse=True)
+            # https://github.com/numpy/numpy/blob/03b62604eead0f7d279a5a4c094743eb29647368/numpy/lib/arraysetops.py#L358C8-L358C8 # noqa: E501
+            order1 = np.argsort(ar1)
+            aux = ar1[order1]
+            mask = np.empty(aux.shape, dtype=np.bool_)
+            mask[:1] = True
+            mask[1:] = aux[1:] != aux[:-1]
+            ar1 = aux[mask]
+            imask = np.cumsum(mask) - 1
+            inv_idx = np.empty(mask.shape, dtype=np.intp)
+            inv_idx[order1] = imask
+            ar2 = np.unique(ar2)
+
+        ar = np.concatenate((ar1, ar2))
+        # We need this to be a stable sort, so always use 'mergesort'
+        # here. The values from the first array should always come before
+        # the values from the second array.
+        order = ar.argsort(kind='mergesort')
+        sar = ar[order]
+        flag = np.empty(sar.size, np.bool_)
+        if invert:
+            flag[:-1] = (sar[1:] != sar[:-1])
+        else:
+            flag[:-1] = (sar[1:] == sar[:-1])
+        flag[-1:] = invert
+        ret = np.empty(ar.shape, dtype=np.bool_)
+        ret[order] = flag
+
+        # return ret[:len(ar1)]
+        if assume_unique:
+            return ret[:len(ar1)]
+        else:
+            return ret[inv_idx]
+
+    return np_in1d_impl
+
+
+@overload(np.isin)
+def jit_np_isin(element, test_elements, assume_unique=False, invert=False,
+                kind=None):
+    if not (type_can_asarray(element) or type_can_asarray(test_elements)):
+        raise TypingError('isin: first two args must be array-like')
+    if not (isinstance(assume_unique, (types.Boolean, bool))):
+        raise TypingError('isin: Argument "assume_unique" must be boolean')
+    if not (isinstance(invert, (types.Boolean, bool))):
+        raise TypingError('isin: Argument "invert" must be boolean')
+
+    if not (kind is None or isinstance(kind, types.NoneType)):
+        kindval = getattr(kind, "literal_value", kind)
+        if not isinstance(kindval,
+                          (types.UnicodeType, str, types.StringLiteral)):
+            raise TypingError('in1d: Argument "kind" must be a string or None')
+        elif kindval != "sort":
+            raise NumbaValueError('in1d: Only kind="sort" is supported')
+
+    # https://github.com/numpy/numpy/blob/03b62604eead0f7d279a5a4c094743eb29647368/numpy/lib/arraysetops.py#L889 # noqa: E501
+    def np_isin_impl(element, test_elements, assume_unique=False, invert=False,
+                     kind=None):
+
+        if kind is not None and kind != "sort":
+            raise ValueError('isin: Only kind="sort" is supported')
```

Comment:
If `kind=None`, NumPy will use `table` by default, which is not supported in this PR. Change the clause to something like this:

```suggestion
        if kind != "sort":
            raise ValueError('isin: Only kind="sort" is supported')
```

This will force a user to spell the `kind="sort"` keyword argument and not assume Numba supports the `table` option.

Reply:
> If kind=None, NumPy will use table by default, which is not supported in this PR.

Yes, that is why I didn't implement kind=None in the first place. I'll change it back.

Reply:
The function signature needs to be with `kind=None` as it is what NumPy specifies. But let's keep a check to only run the function if the user specifies `kind="sort"`.

Let me know if you have any questions regargind this.